svgdigitizer/electrochemistry/cv.py

```diff
@@ -19,110 +19,199 @@
 #  You should have received a copy of the GNU General Public License
 #  along with svgdigitizer. If not, see <https://www.gnu.org/licenses/>.
 # ********************************************************************
+from functools import cache
 import pandas as pd
 import matplotlib.pyplot as plt
 from pathlib import Path
+from astropy import units as u
 
 
 class CV():
-    def __init__(self, metadata, svgplot):
-        """
-        metadata: dict
-        """
+    # TODO: Add documentation with a usage example. #60
+    def __init__(self, svgplot, metadata=None):
         self.svgplot = svgplot
-        self.metadata = metadata
-        self.svgplot.create_df()
+        self._metadata = metadata or {}
 
-        # TODO: These labels should either be extracted from the svg or yaml file
-        self.xlabel = 'U'
+    @property
+    @cache
+    def axis_properties(self):
+        return {'x': {'dimension': 'U',
+                      'unit': 'V'},
+                'y': {'dimension': 'j' if 'm2' in str(CV.get_axis_unit(self.svgplot.units['x'])) else 'I',
+                      'unit': 'A / m2' if 'm2' in str(CV.get_axis_unit(self.svgplot.units['y'])) else 'A'}}
 
-        # TODO: should be j if both current and normalized to are given in the yaml file
-        self.ylabel = 'I'
+    @classmethod
+    def get_axis_unit(cls, unit):
+        r"""
+        Convert a string containing a unit into a string that can be used by the astropy.unit module.
+        For example 'uA cm-2' should become 'uA / cm2'
```

Comment:
That's not true actually. It does not return a string but a unit.
```suggestion
        Return `unit` as an astropy unit.
        
        This method normalizes unit names, e.g., it rewrites 'uA cm-2' to 'uA / cm2' which astropy understands.
```